policy/modules/system/systemd.if

```diff
@@ -58,28 +60,612 @@ template(`systemd_role_template',`
 	# This domain is per-role because of the below transitions.
 	# See the systemd --user section of systemd.te for the
 	# remainder of the rules.
-	allow $1_systemd_t $3:process { setsched rlimitinh };
+	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
+	allow $1_systemd_t self:netlink_kobject_uevent_socket create_socket_perms;
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
-	allow $1_systemd_t self:process signal;
+	allow $1_systemd_t self:process { getsched signal };
+
+	dontaudit $3 $1_systemd_t:unix_stream_socket rw_stream_socket_perms;
 
 	files_search_home($1_systemd_t)
 
+	init_link_keyring($1_systemd_t)
+
+	miscfiles_watch_localization($1_systemd_t)
+
+	mount_read_runtime_files($1_systemd_t)
+	mount_watch_runtime_files($1_systemd_t)
+	mount_watch_reads_runtime_files($1_systemd_t)
+
+	allow $3 $1_systemd_t:process signal;
 	# Allow using file descriptors for user environment generators
 	allow $3 $1_systemd_t:fd use;
 	allow $3 $1_systemd_t:fifo_file rw_inherited_fifo_file_perms;
 
+	allow $1_systemd_t self:unix_stream_socket rw_socket_perms;
+
+	allow $3 $1_systemd_t:system { disable enable reload start stop status };
+
 	# systemctl --user
 	stream_connect_pattern($3, systemd_user_runtime_t, systemd_user_runtime_t, $1_systemd_t)
 
+	allow $1_systemd_t $3:dir search_dir_perms;
+	allow $1_systemd_t $3:file read_file_perms;
+	allow $1_systemd_t $3:lnk_file read_lnk_file_perms;
+
 	can_exec($3, { systemd_run_exec_t systemd_analyze_exec_t })
 
 	dbus_system_bus_client($1_systemd_t)
+	dbus_spec_session_bus_client($1, $1_systemd_t)
+	dbus_connect_spec_session_bus($1, $1_systemd_t)
+	dbus_manage_session_dbus_runtime_sock_files($1_systemd_t)
```

Comment:
Yes but this does not really scale well and you could use type attributes for this:

```
allow $1_systemd_t systemd_user_unix_stream_activated_socket_type:unix_stream_socket { create_socket_perms listen };
allow $1_systemd_t systemd_user_activated_sock_file_type:dir manage_dir_perms;
allow $1_systemd_t systemd_user_activated_sock_file_type:dir manage_sock_file_perms;
```
Then create interfaces like so

```
interface(`systemd_user_activated_sock_file`,'
    gen_require(` attribute systemd_user_activated_sock_file_type; ')
    type_attribute $1 systemd_user_activated_sock_file_type;
')
interface(`systemd_user_unix_stream_socket_activated`,'
    gen_require(` attribute systemd_user_unix_stream_socket_activated_type; ')
    type_attribute $1 systemd_user_unix_stream_socket_activated_type;
    systemd_user_activated_sock_file($2)
')
```
then call that instead in , i guess the dbus role:
```
systemd_user_unix_stream_socket_activated($1_dbusd_t, dbus_session_runtime_t)
```

Reply:
OK. That makes more sense.